src/app/forms/custom-sections/body/body.component.ts

```diff
@@ -68,7 +74,18 @@ export class BodyComponent implements OnInit, OnDestroy {
     return this.optionsService.getOptions(ReferenceDataResourceType.BodyMake);
   }
 
+  get dtpNumbers$(): Observable<MultiOptions> {
+    return this.optionsService.getOptions(ReferenceDataResourceType.PsvMake).pipe(
+      skipWhile(data => data?.length! <= 2000), //// Write better predicate
```

Comment:
Can anyone help here? This is really the best I could do. Need to wait until the whole result is in the store basically here, but the observable doesn't 'close' so you cannot just like get the last once it's done. This was my soloution but basically need something that says 'wait until pagination has ended and you have all 2.8K responses'. This currently works but won't protect if the ref data increases

Reply:
Or create a selector just to get the loading flag of PsvMake store feature

Reply:
Or a selector that takes the name of a store feature in reference data and returns the loading state of that feature

Reply:
Why do you need it to wait for all to be there, won't the consumer just react to the changes in state?

Reply:
Yeah I think this will work, really good idea!

Reply:
If you need to wait then you could combine loading flag and data into one observable. Then skip while loading is true. In the effect and reducer, the effect emits `paginated` which sets the loading flag in the reducer. Finally, when no more pagination token, loading is set to false.

Reply:
In answer to the 'why do you need to wait' bit, it appears that the autocomplete does not take in changes once it loads. There is this line `takeWhile(options => !options || options.length === 0, true)` which means that once it has any data, it kills the observable, so this observable emits after first pagination, spits it down, and then this line shuts off so it never updates.

Reply:
Sorry I was looking at a stale version of the page, didn't know @gjulien-bjss had already given the same suggestion 😂